File: 0507. Perfect Number/Solution.py
class Solution:
    def checkPerfectNumber_MK1(self, num: int) -> bool:
        if num <= 1:
            return False
        accum = 1
        for i in range(2, int(num ** 0.5) + 1):
            if num % i == 0:
                accum += i + num // i
        return accum == num

    # Euclid-Euler Theorem
    def checkPerfectNumber_MK2(self, num: int) -> bool:
        for i in [2, 3, 5, 7, 13]:
            if num == (2 ** (i - 1)) * (2 ** i - 1):
                return True
        return False

File: 0507. Perfect Number/Solution.java
class Solution {
    public boolean checkPerfectNumber_MK1(int num) {
        if (num <= 1)
            return false;
        int sum = 1;
        for (int i = 2; i * i <= num; i++)
            if (num % i == 0)
                sum += i + num / i;
        return sum == num;
    }

    // Euclid-Euler Theorem
    public boolean checkPerfectNumber_MK2(int num) {
        int[] primes = {2, 3, 5, 7, 13};
        for (int prime : primes) {
            int t = Math.pow(2, prime - 1) * (Math.pow(2, prime) - 1);
            if (t == num)
                return true;
        }
        return false;
    }
}
